indico/modules/rb_new/client/js/util.js

```diff
@@ -21,3 +21,32 @@ import moment from 'moment';
 export function toMoment(dt, format) {
     return dt ? moment(dt, format) : null;
 }
+
+export function parseRoomListFiltersText(text) {
+    const result = {text: null, building: null, floor: null};
+    if (!text) {
+        return result;
+    }
+
+    const reSplit = /((?:building:|floor:)(?:.*?)(?: |$))/;
```

Comment:
shouldn't we do something more generic here? ie split in space-separated tokens, and then see if it's matching `foo:bar` - if not it's just a normal part of the search string, otherwise you add it as `[key]: value` to the result dict